### DISCUSSION TITLE
A "batteries not included" project type?

### DISCUSSION THREAD

**Discussion by vaughandroid** [projen/projen#2993] [Q&A] [answered]
I like the basic idea of projen, but its "batteries included" approach looks likely to be a serious barrier to adoption for my team.

We use Node and TypeScript, but many of our choices for tools and libraries are different from those in the standard `TypeScriptProject`. So far, my .projenrc.ts file is more about what my project _doesn't_ do, or need, rather than what it _does_.

I know the docs recommend that we "take the time to fully migrate to the expected project structure", but that feels like a case of the tail wagging the dog - we're happy with most of our current tools/libraries. The problem I'm trying to solve here is that of keeping dozens of projects aligned in a mature monorepo.

It looks like the cleanest option might be to start with a simple `Project`, and build ourselves something custom from there.

Is that the best option, or are there better alternatives?

**Comment by mrgrain** [comment, marked as answer]
> It looks like the cleanest option might be to start with a simple Project, and build ourselves something custom from there.

Yeah, if your implementation is really so different from the "batteries included" ones, I think that's the best starting point. Most Components should be reuseable in custom projects.
